native/packages/react-native-bpk-component-icon/src/BpkIcon.js

```diff
@@ -42,7 +44,14 @@ const styles = StyleSheet.create({
 const mapCharacterCode = characterCode =>
   String.fromCharCode(parseInt(characterCode, 16));
 
-const BpkIcon = props => {
+type IconName = $Keys<typeof iconMappings>;
+type Props = {
+  icon: IconName,
+  small: ?boolean,
+  style: ?{} | Array<{}>,
```

Comment:
This is probably going to become a common pattern, so we could abstract it out. Alternatively, we could pull a type out of `react-native` itself. For now, this works well.

Reply:
How does React Native itself do these?